contrib/pkg/certificate/create.go

```diff
@@ -0,0 +1,174 @@
+package certificate
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/route53"
+
+	awsclient "github.com/openshift/hive/pkg/awsclient"
+)
+
+// Options is the set of options to create a new server certificate
+type Options struct {
+	Name       string
+	BaseDomain string
+	Email      string
+	Region     string
+	OutputDir  string
+	WaitTime   time.Duration
+}
+
+// NewCreateCertifcateCommand returns a command that will create a letsencrypt serving cert
+// for OpenShift clusters on AWS.
+func NewCreateCertifcateCommand() *cobra.Command {
+	username := "user"
+	homeDir := "/tmp"
+	user, err := user.Current()
+	if err == nil {
+		username = user.Username
+		homeDir = user.HomeDir
+	}
+
+	opt := &Options{}
+	cmd := &cobra.Command{
+		Use:   "create CLUSTER_NAME",
+		Short: "Creates a new serving certificate for the given cluster",
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := opt.Complete(cmd, args); err != nil {
+				return
+			}
+			log.SetLevel(log.InfoLevel)
+			err := opt.Run()
+			if err != nil {
+				log.WithError(err).Error("Error")
+			}
+		},
+	}
+	flags := cmd.Flags()
+	flags.StringVar(&opt.BaseDomain, "base-domain", "new-installer.openshift.com", "Base domain for the cluster")
+	flags.StringVar(&opt.Email, "email", fmt.Sprintf("%s@redhat.com", username), "email address to use for certificate")
+	flags.StringVar(&opt.Region, "region", "us-east-1", "AWS region for certificate hosted zone")
+	flags.StringVar(&opt.OutputDir, "output-dir", homeDir, "Output directory for certs. Defaults to home directory")
+	flags.DurationVar(&opt.WaitTime, "wait-time", 30*time.Second, "amount of time to wait after creating dns entries so that they can be authenticated")
+	return cmd
+}
+
+// Complete finalizes command options
+func (o *Options) Complete(cmd *cobra.Command, args []string) error {
+	if len(args) != 1 {
```

Comment:
Set the `Args` field in the `cobra.Command` object to `ExactArgs(1)`. That will take care of this check for you.